test/integration/node-specific/bson-options/useBigInt64.test.ts

```diff
@@ -0,0 +1,234 @@
+      expect(client.s.bsonOptions.useBigInt64).to.exist;
+      expect(client.s.bsonOptions.useBigInt64).to.be.false;
+    });
+  });
+
+  describe('when set at client level', function () {
+    let client: MongoClient;
+    beforeEach(function () {
+      client = configuration.newClient(configuration.writeConcernMax(), {
+        useBigInt64: true
+      });
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it('supercedes driver level', function () {
+      expect(client.s.bsonOptions.useBigInt64).to.exist;
+      expect(client.s.bsonOptions.useBigInt64).to.be.true;
+    });
+  });
+
+  describe('when set at DB level', function () {
+    let client: MongoClient;
+    let db: Db;
+    beforeEach(async function () {
+      client = configuration.newClient(configuration.writeConcernMax(), {
+        useBigInt64: false
+      });
+
+      db = client.db(configuration.db, { useBigInt64: true });
+      await db.dropCollection('useBigInt64Test').catch(() => null);
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it('supercedes client level', async function () {
+      expect(db.s.bsonOptions.useBigInt64).to.exist;
+      expect(db.s.bsonOptions.useBigInt64).to.be.true;
+    });
+  });
+
+  describe('when set at collection level', function () {
+    let client: MongoClient;
+    let db: Db;
+    let col: Collection;
+
+    beforeEach(async function () {
+      client = configuration.newClient(configuration.writeConcernMax());
+
+      db = client.db(configuration.db, { useBigInt64: false });
+      await db.dropCollection('useBigInt64Test').catch(() => null);
+      col = await db.createCollection('useBigInt64Test', { useBigInt64: true });
+    });
+
+    afterEach(async function () {
+      await db
+        .dropCollection('useBigInt64Test')
+        .catch(() => expect.fail('failed to drop collection'));
+      await client.close();
+    });
+
+    it('supercedes db level', function () {
+      expect(col.s.bsonOptions.useBigInt64).to.exist;
+      expect(col.s.bsonOptions.useBigInt64).to.be.true;
+    });
+  });
+
+  describe('when set at operation level', function () {
+    let client: MongoClient;
+    let db: Db;
+    let col: Collection;
+
+    beforeEach(async function () {
+      client = configuration.newClient(configuration.writeConcernMax());
+
+      db = client.db(configuration.db);
+      await db.dropCollection('useBigInt64Test').catch(() => null);
+
+      col = await db.createCollection('useBigInt64Test');
+      await col.insertMany([{ a: 1n }, { a: 2n }, { a: 3n }, { a: 4n }]);
+    });
+
+    afterEach(async function () {
+      await db
+        .dropCollection('useBigInt64Test')
+        .catch(() => expect.fail('failed to drop collection'));
+      await client.close().catch(() => expect.fail('failed to close client'));
+    });
+
+    it('deserializes to a bigint', async function () {
```

Comment:
This isn't right.  What we expect is that it is possible to override more global `useBigInt64` configurations when we set it on the operation.  We don't always expect it to enable bigint deserialization.

So this test should be similar to the others. Except that we need to round trip information to confirm that the values are what we expect instead of inspecting options.  Alternatively, we could spy on the call to serialize and confirm that the option is set.  My preference would be the former

Reply:
Would it suffice to explicitly set the global option to false and change it to true at the operation level, then check that the option made its way through by checking the type of the result in addition to doing it the other way around (globally set to true, set to false at the operation level)?